src/cpp.cc

```diff
@@ -21,48 +21,44 @@
 
 }}} */
 
+#include "cpp.hh"
+
 #include <cmath>
+#include <list>
 #include <set>
+#include <string>
 #include <tuple>
 #include <utility>
 #include <vector>
-#include <string>
-#include <list>
 #include <boost/tokenizer.hpp>
 
-#include "cpp.hh"
-
+#include "ast.hh"
+#include "cyk.hh"
+#include "expr.hh"
+#include "expr/new.hh"
+#include "fn_def.hh"
+#include "grammar.hh"
+#include "options.hh"
+#include "outside/codegen.hh"
 #include "statement.hh"
 #include "statement/backtrace_decl.hh"
+#include "statement/fn_call.hh"
 #include "statement/hash_decl.hh"
```

Comment:
This is a bit confusing. Why did you reorder the imports? Current tests break because of some import issues, is this here the cause? Hard to debug. Could you please revert as many lines as possible to show what really changed?